library/pkcs5.c

```diff
@@ -230,9 +231,13 @@ int mbedtls_pkcs5_pbkdf2_hmac( mbedtls_md_context_t *ctx, const unsigned char *p
 
     memset( counter, 0, 4 );
     counter[3] = 1;
-
+/*
+ * relevant only for 64 bit platforms. On 32 bit platforms, unsigned int cannot exceed 32 bits
+ */
+#if defined(MBEDTLS_HAVE_INT64)
     if( iteration_count > 0xFFFFFFFF )
```

Comment:
@andresag01 Since unsigned int is compiler dependent, and there could be some compilers that unsigned int might be 64 bits, I think it is better to keep this